user_guide_src/source/general/view_parser.rst

```diff
@@ -2,5 +2,382 @@
+data... {/blog_entries}. In a case like this, the entire chunk of data
+between these pairs would be repeated multiple times, corresponding to
+the number of rows in the "blog_entries" element of the parameters array.
+
+Parsing variable pairs is done using the identical code shown above to
+parse single variables, except, you will add a multi-dimensional array
+corresponding to your variable pair data. Consider this example::
+
+	$data = array(
+		'blog_title'   => 'My Blog Title',
+		'blog_heading' => 'My Blog Heading',
+		'blog_entries' => array(
+			array('title' => 'Title 1', 'body' => 'Body 1'),
+			array('title' => 'Title 2', 'body' => 'Body 2'),
+			array('title' => 'Title 3', 'body' => 'Body 3'),
+			array('title' => 'Title 4', 'body' => 'Body 4'),
+			array('title' => 'Title 5', 'body' => 'Body 5')
+		)
+	);
+
+	echo $parser->setData($data)
+		->render('blog_template');
+
+If your "pair" data is coming from a database result, which is already a
+multi-dimensional array, you can simply use the database ``getResultArray()``
+method::
+
+	$query = $db->query("SELECT * FROM blog");
+
+	$data = array(
+		'blog_title'   => 'My Blog Title',
+		'blog_heading' => 'My Blog Heading',
+		'blog_entries' => $query->getResultArray()
+	);
+
+	echo $parser->setData($data)
+		->render('blog_template');
+
+Variable Pairs for Nested Substitutions
+=======================================
+
+TODO
+
+Conditional Substitutions
+=========================
+
+TODO
+
+***********
+Usage Notes
+***********
+
+If you include substitution parameters that are not referenced in your
+template, they are ignored::
+
+	$template = 'Hello, {firstname} {lastname}';
+	$data = array(
+		'title' => 'Mr',
+		'firstname' => 'John',
+		'lastname' => 'Doe'
+	);
+	echo $parser->setData($data)
+		->renderString($template);
+
+	// Result: Hello, John Doe
+
+If you do not include a substitution parameter that is referenced in your
+template, the original pseudo-variable is shown in the result::
+
+	$template = 'Hello, {firstname} {initials} {lastname}';
+	$data = array(
+		'title' => 'Mr',
+		'firstname' => 'John',
+		'lastname' => 'Doe'
+	);
+	echo $parser->setData($data)
+		->renderString($template);
+
+	// Result: Hello, John {initials} Doe
+
+If you provide a string substitution parameter when an array is expected,
+i.e. for a variable pair, the substitution is done for the opening variable
+pair tag, but the closing variable pair tag is not rendered properly::
+
+	$template = 'Hello, {firstname} {lastname} ({degrees}{degree} {/degrees})';
+	$data = array(
+		'degrees' => 'Mr',
+		'firstname' => 'John',
+		'lastname' => 'Doe',
+		'titles' => array(
+			array('degree' => 'BSc'),
+			array('degree' => 'PhD')
+		)
+	);
+	echo $parser->setData($data)
+		->renderString($template);
+
+	// Result: Hello, John Doe (Mr{degree} {/degrees})
+
+If you name one of your individual substitution parameters the same as one
+used inside a variable pair, the results may not be as expected::
```

Comment:
Unclear. Wait until I get it a bit better fleshed out. I am working on a definitive description of the substitution order and behaviors! I am not purposefully trying to make it behave the same as the CI3 one.